pkg/agent/syssetup/setup.go

```diff
@@ -39,12 +43,50 @@ func Configure(enableIPv6 bool) {
 	// Kernel is inconsistent about how devconf is configured for
 	// new network namespaces between ipv4 and ipv6. Make sure to
 	// enable forwarding on all and default for both ipv4 and ipv6.
-	enableSystemControl("/proc/sys/net/ipv4/conf/all/forwarding")
-	enableSystemControl("/proc/sys/net/ipv4/conf/default/forwarding")
-	enableSystemControl("/proc/sys/net/bridge/bridge-nf-call-iptables")
+	sysctls := map[string]int{
+		"net/ipv4/conf/all/forwarding":       1,
+		"net/ipv4/conf/default/forwarding":   1,
+		"net/bridge/bridge-nf-call-iptables": 1,
+		"net/netfilter/nf_conntrack_max":     getConntrackMax(),
+	}
+
 	if enableIPv6 {
-		enableSystemControl("/proc/sys/net/ipv6/conf/all/forwarding")
-		enableSystemControl("/proc/sys/net/ipv6/conf/default/forwarding")
-		enableSystemControl("/proc/sys/net/bridge/bridge-nf-call-ip6tables")
+		sysctls["net/ipv6/conf/all/forwarding"] = 1
+		sysctls["net/ipv6/conf/default/forwarding"] = 1
+		sysctls["net/bridge/bridge-nf-call-ip6tables"] = 1
+	}
+
+	sys := sysctl.New()
+	for entry, value := range sysctls {
+		if val, _ := sys.GetSysctl(entry); val != value {
+			logrus.Infof("Set sysctl '%v' to %v", entry, value)
+			if err := sys.SetSysctl(entry, value); err != nil {
+				logrus.Errorf("Failed to set sysctl: %v", err)
+			}
+		}
+	}
+}
+
+// getConntrackMax is cribbed from kube-proxy, as recent kernels no longer allow non-init namespaces
+// to set conntrack-related sysctls. Rather than promote the conntrack-max-per-core and
+// conntrack-min kube-proxy CLI flags up to top-level flags, or parse kube-proxy args, we just
+// hardcode the defaults.
+func getConntrackMax() int {
+	scaled := int(ConntrackMaxPerCore) * detectNumCPU()
+	if scaled > ConntrackMin {
+		logrus.Debugf("getConntrackMax: using scaled ConntrackMaxPerCore")
+		return scaled
+	}
+	logrus.Debugf("getConntrackMax: using ConntrackMin")
+	return ConntrackMin
+}
+
+// detectNumCPU is also cribbed from kube-proxy
+func detectNumCPU() int {
+	// try get numCPU from /sys firstly due to a known issue (https://github.com/kubernetes/kubernetes/issues/99225)
+	_, numCPU, err := machine.GetTopology(sysfs.NewRealSysFs())
+	if err != nil || numCPU < 1 {
```

Comment:
Are we sure we want to swallow this error? The lib itself calls a number of supporting funcs and returns any number of errors.

Reply:
this is what kube-proxy does so apparently it's good enough for Kubernetes!

Reply:
That's, well, ... ugh...